packages/client/src/runtime/core/types/GetResult.ts

```diff
@@ -48,13 +48,12 @@ export type FluentOperation =
 type Count<O> = { [K in keyof O]: Count<number> } & {}
 
 // prettier-ignore
-export type GetFindResult<P extends Payload, A> =
-  {} extends A ? DefaultSelection<P> :
-  A extends 
+export type GetFindResult<P extends Payload, _A, A = { 0: _A, 1: { select: undefined } }[Equals<_A, any>]> =
+  A extends
```

Comment:
The removed condition `{} extends A ? DefaultSelection<P> :` caused a very huge performance hit. Instead we now have an ugly comparison to take some steps when args `A` equals `any`, `{}` worked but was too wide-matching, I guess.